include/transmem/transmem.h

```diff
@@ -59,41 +60,64 @@ class EmptyLinkQueryException : public std::runtime_error {
     Link emptyLink;
 };
 
-
-/****************************
- * PATH                     *
- ****************************/
+/********
+ * PATH *
+ ********/
 
 struct Path {
+
     FrameID src;
     FrameID dst;
     std::vector< std::reference_wrapper<Link> > links;
 
     void writeJSON(QJsonObject &json) const;
 };
 
+/************************************
+ * STAMPED AND RATED TRANSFORAMTION *
+ * **********************************/
+
+/* REMARK:
+   New return type of transmem. Encodes not just the transformation but also
+   some information about the quality. Unnecessary convertion between matrix and quaternion
+   are no longer needed and the return type is much more flexible, one can add additional
+   stuff without changing interface to much. */
+
+struct StampedAndRatedTransformation {
```

Comment:
I can do that. During the implementation i realized that i need the rotation and the translation part of the transformation separately. I don't wanted to convert unnecessary between 4x4 matrices back and forth.